### PR TITLE
run Lighthouse audit on CI

### LINKED ISSUE #71: run lighthouse audit on PRs
Run Lighthouse audit on PRs.

# Useful Articles

- [Your first performance budget with lighthouse](https://bitsofco.de/your-first-performance-budget-with-lighthouse/)
- [Performant front-end architecture](https://www.debugbear.com/blog/performant-front-end-architecture)
- [Introduction to running lighthouse programmatically](https://www.smashingmagazine.com/2020/09/introduction-running-lighthouse-programmatically/)
- [2021 front-end performance checklist](https://www.smashingmagazine.com/2021/01/front-end-performance-2021-free-pdf-checklist/)
- https://speedcurve.com/blog/performance-budgets-guide/?utm_source=CSS-Weekly&utm_campaign=Issue-451&utm_medium=web
- https://web.dev/optimize-vitals-lighthouse/

### PR DESCRIPTION
**Pull request by dwhieb**
**Related Issue:**

closes #71

**Description of Changes**

Runs a [Lighthouse](https://developers.google.com/web/tools/lighthouse/) audit on PRs and commits to `main`, checking to make sure we don't exceed performance budgets.

Audits that were original failing are currently disabled. I opened #243 as a reminder to make issues to address these disabled audits (as appropriate).

**NOTE:** Documentation for using Lighthouse has been added to #240 rather than this PR.

### CODE REVIEW

**Review by vadekh — COMMENTED** [review]
@dwhieb when I run `npm run test` or just `npm run test:perf` the command does not terminate by itself (unless it is meant to take 4+ minutes). The following looks like the job should be completed? But let me know if I just need to let it run longer...
Screen Shot 2021-09-27 at 11 05 04 AM

**Comment by vadekh** [comment]
Also, looks like lighthouse generates some files. They can probably end up in `.gitignore`. An `assertion-results.json` file, and these:
Screen Shot 2021-09-27 at 11 23 41 AM

**Comment by dwhieb** [comment]
> Also, looks like lighthouse generates some files. They can probably end up in `.gitignore`.

Oops, I added that folder to `.gitignore` but on the wrong branch. Fixed.

**Comment by dwhieb** [comment]
> @dwhieb when I run `npm run test` or just `npm run test:perf` the command does not terminate by itself

It terminates on my Windows machine, so this might be a platform difference. Does using `&&` instead of `&` in the npm script make any difference?

What happens if you run this directly on the command line?

`npx lhci autorun && npx rimraf .lighthouseci`

**Comment by vadekh** [comment]
`&&` instead of `&` fixes the termination issue on my end, in both the `test` and `test:perf` commands

File: package.json [hunk 1]
```diff
@@ -32,16 +32,18 @@
     "cypress-run": "cypress run --config-file test/cypress.json",
     "start": "node build/bin/startServer.js",
     "stop-only": "stop-only -f test,src",
-    "test": "npm run test:unit & npm run test:integration",
+    "test": "npm run test:unit & npm run test:integration & npm run test:perf",
     "test:component": "cypress run-ct --config-file test/cypress.json",
     "test:integration": "start-server-and-test start 3000 cypress-run",
+    "test:perf": "lhci autorun & rimraf .lighthouseci",
```
Comment (vadekh) [inline package.json]:
```suggestion
    "test:perf": "lhci autorun && rimraf .lighthouseci",
```

File: package.json [hunk 2]
```diff
@@ -32,16 +32,18 @@
     "cypress-run": "cypress run --config-file test/cypress.json",
     "start": "node build/bin/startServer.js",
     "stop-only": "stop-only -f test,src",
-    "test": "npm run test:unit & npm run test:integration",
+    "test": "npm run test:unit & npm run test:integration & npm run test:perf",
```
Comment (vadekh) [inline package.json]:
```suggestion
    "test": "npm run test:unit && npm run test:integration && npm run test:perf",
```

**Review by vadekh — CHANGES_REQUESTED** [review]

**Review by vadekh — APPROVED** [review]

**Pull request merged**